lua/neotest-dotnet/framework-discovery.lua

```diff
@@ -96,15 +96,24 @@ function M.get_test_framework_utils_from_source(source, custom_attribute_args)
       )
   ]]
 
+  local fsharp_query = [[
+        (attribute 
+          (simple_type 
+            (long_identifier (identifier) @attribute_name (#any-of? @attribute_name ]] .. xunit_attributes .. " " .. nunit_attributes .. " " .. mstest_attributes .. [[)))
+        )
+    ]]
+
+  local framework_query = lang == "fsharp" and fsharp_query or c_sharp_query
+
   async.scheduler()
-  local root = vim.treesitter.get_string_parser(source, "c_sharp"):parse()[1]:root()
+  local root = vim.treesitter.get_string_parser(source, lang):parse()[1]:root()
   local parsed_query = vim.fn.has("nvim-0.9.0") == 1
-      and vim.treesitter.query.parse("c_sharp", framework_query)
-    or vim.treesitter.parse_query("c_sharp", framework_query)
-  for _, captures in parsed_query:iter_matches(root, source) do
+      and vim.treesitter.query.parse(lang, framework_query)
+    or vim.treesitter.parse_query(lang, framework_query)
+  for _, node, _, _ in parsed_query:iter_captures(root, source) do
```

Comment:
Not sure if this changes the behavior of not, but on nightly `iter_matches` have been changed/fixed to return all matches instead of only the last one.